source/common/upstream/outlier_detection_impl.h

```diff
@@ -205,6 +217,15 @@ class DetectorHostMonitorImpl : public DetectorHostMonitor {
   void putResultWithLocalExternalSplit(Result result, absl::optional<uint64_t> code);
   std::function<void(DetectorHostMonitorImpl*, Result, absl::optional<uint64_t> code)>
       put_result_func_;
+
+  // Set of extension monitors.
+  std::unique_ptr<ExtMonitorsSet> monitors_set_;
```

Comment:
How about to use `ExtMonitorsSet monitors_set_;` directly? I don't think the unique_ptr here make sense.

Reply:
Good idea. Refactored as you suggested. This probably also improved performance. Thanks!